main/exercise/ptest_agree_reorder.class.php

```diff
@@ -0,0 +1,293 @@
+                $defaults['ptest_category['.$i.']'] = isset($answer->ptest_category[$i]) ? $answer->ptest_category[$i] : 0;
+            }
+
+            $renderer = $form->defaultRenderer();
+            $renderer->setElementTemplate(
+                '<td><!-- BEGIN error --><span class="form_error">{error}</span><!-- END error --><br/>{element}</td>',
+                'counter['.$i.']'
+            );
+            $renderer->setElementTemplate(
+                '<td><!-- BEGIN error --><span class="form_error">{error}</span><!-- END error --><br/>{element}</td>',
+                'answer['.$i.']'
+            );
+            $renderer->setElementTemplate(
+                '<td><!-- BEGIN error --><span class="form_error">{error}</span><!-- END error --><br/>{element}</td>',
+                'ptest_category['.$i.']'
+            );
+            $answer_number = $form->addElement(
+                'text',
+                'counter['.$i.']',
+                null,
+                ' value = "'.$i.'"'
+            );
+            $answer_number->freeze();
+
+            $form->addHtmlEditor('answer['.$i.']', null, null, false, $editor_config);
+
+            $form->addRule(
+                    'answer['.$i.']',
+                    get_lang('ThisFieldIsRequired'),
+                    'required'
+                    );
+
+            $form->addSelect(
+                'ptest_category['.$i.']',
+                null,
+                $categoriesOptions
+            );
+
+            $form->addHtml('</tr>');
+        }
+
+        $form->addHtml('</tbody>');
+        $form->addHtml('</table>');
+
+        global $text;
+        $buttonGroup = [];
+
+        if ($obj_ex->edit_exercise_in_lp == true ||
+            (empty($this->exerciseList) && empty($obj_ex->id))
+        ) {
+            //setting the save button here and not in the question class.php
+            $buttonGroup[] = $form->addButtonDelete(get_lang('LessAnswer'), 'lessAnswers', true);
+            $buttonGroup[] = $form->addButtonCreate(get_lang('PlusAnswer'), 'moreAnswers', true);
+            $buttonGroup[] = $form->addButton(
+                    'submitQuestion',
+                    $text,
+                    'check',
+                    'primary',
+                    'default',
+                    null,
+                    ['id' => 'submit-question'],
+                    true
+                    );
+            $form->addGroup($buttonGroup);
+        }
+
+        if (!empty($this->id)) {
+            $form->setDefaults($defaults);
+        } else {
+            if ($this->isContent == 1) {
+                // Default sample content.
+                $form->setDefaults($defaults);
+            }
+        }
+        $form->setConstants(['nb_answers' => $nb_answers]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function processAnswersCreation($form, $exercise)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $nb_answers is not named in camelCase.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)